src/libecalc/application/energy_calculator.py

```diff
@@ -41,39 +44,82 @@
+        if len(consumer_power_usage) == 1:
+            return consumer_power_usage[0]
+
+        return reduce(operator.add, consumer_power_usage)
+
+    def get_fuel_usage(self) -> Optional[TimeSeriesStreamDayRate]:
+        energy_usage = self._consumer_results[self._component_id].component_result.energy_usage
+        if energy_usage.unit == Unit.MEGA_WATT:
+            # energy usage is power usage, not fuel usage.
+            return None
+        return energy_usage
+
+
 class EnergyCalculator:
     def __init__(
         self,
-        graph: ComponentGraph,
+        energy_model: EnergyModel,
+        expression_evaluator: ExpressionEvaluator,
     ):
-        self._graph = graph
+        self._energy_model = energy_model
+        self._expression_evaluator = expression_evaluator
+        self._consumer_results: dict[str, EcalcModelResult] = {}
 
-    def evaluate_energy_usage(self, variables_map: VariablesMap) -> dict[str, EcalcModelResult]:
-        component_ids = list(reversed(self._graph.sorted_node_ids))
-        component_dtos = [self._graph.get_node(component_id) for component_id in component_ids]
+    def _get_context(self, component_id: str) -> ComponentEnergyContext:
+        return Context(
+            energy_model=self._energy_model,
+            consumer_results=self._consumer_results,
+            component_id=component_id,
+        )
 
-        consumer_results: dict[str, EcalcModelResult] = {}
+    def evaluate_energy_usage(self) -> dict[str, EcalcModelResult]:
+        energy_components = self._energy_model.get_energy_components()
 
-        for component_dto in component_dtos:
-            if isinstance(component_dto, ElectricityConsumerDTO | FuelConsumerDTO):
+        for energy_component in energy_components:
+            if isinstance(energy_component, ElectricityConsumerDTO | FuelConsumerDTO):
                 consumer = Consumer(
-                    id=component_dto.id,
-                    name=component_dto.name,
-                    component_type=component_dto.component_type,
-                    regularity=TemporalModel(component_dto.regularity),
-                    consumes=component_dto.consumes,
+                    id=energy_component.id,
+                    name=energy_component.name,
+                    component_type=energy_component.component_type,
+                    regularity=TemporalModel(energy_component.regularity),
+                    consumes=energy_component.consumes,
                     energy_usage_model=TemporalModel(
                         {
                             period: EnergyModelMapper.from_dto_to_domain(model)
-                            for period, model in component_dto.energy_usage_model.items()
+                            for period, model in energy_component.energy_usage_model.items()
                         }
                     ),
                 )
-                consumer_results[component_dto.id] = consumer.evaluate(expression_evaluator=variables_map)
```

Comment:
Next -> move the contents of each if into the corresponding dto in order to not depend on dto in energy calculator. Need to pass expression_evaluator as a parameter